### PR TITLE
GOV-301: Add Authorization worker for Batch Authorization

### PR DESCRIPTION
**Pull request by danishjamal104**
Jira link: [GOV-301](https://mifosforge.jira.com/browse/GOV-301)

Adds 2 workers to bulk processor.

Authorization worker - Calls mock async authorization API in mock payment schema.
Authorization Callback Handler (Message task)
Outcome: Successful authorization with zeebe variable AUTHORIZATION_SUCCESSFUL as true.

### Commit List
* GOV-301: handle callback (incomplete)
* GOV-301: updating the response handling of async api call
* GOV-301: minor code refactor
* GOV-301: applying spotless changes
* testing changes
* GOV-301: updated changes
* GOV-301: add updated s3 details
* GOV-301: updated changes with refactoring
* GOV-301: working and refactored code
* GOV-301: refactored code
* GOV-301: add approved amount zeebe variable
* update url
* GOV-301: addressing review comments
* Fixed compile time error after conflict resolution

[GOV-301]: https://mifosforge.jira.com/browse/GOV-301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/mifos/processor/bulk/api/CallbackController.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Autowired
+    private ZeebeClient zeebeClient;
+
+    protected Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private static final String EXPECTED_AUTH_STATUS = "Y";
+
+
+    @PostMapping("/authorization/callback")
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/api/CallbackController.java:29]: can you build this API as per the REST API standards

Comment (fynmanoj) [reply src/main/java/org/mifos/processor/bulk/api/CallbackController.java:29]: check the URL naming convention

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -182,3 +190,5 @@ payment-mode:
       type: "BULK"
       endpoint: "bulk_connector_{MODE}-{dfspid}"
       debulkingDfspid: "lion"
+batch-authorization:
+  callback-url: "https://bulk-connector.sandbox.fynarfin.io/authorization/callback"
```
Comment (fynmanoj) [inline src/main/resources/application.yaml:194]: use the service name here. with in the cluster it will be able to connect

**Review by fynmanoj — CHANGES_REQUESTED** [review]
REST standard and configuration change

File: src/main/resources/application.yaml (hunk 2 above)
Comment (danishjamal104) [reply src/main/resources/application.yaml:194]: There is no service with this name... what should I do?

File: src/main/java/org/mifos/processor/bulk/api/CallbackController.java (hunk 1 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/processor/bulk/api/CallbackController.java:29]: What should be the endpoint? or Where is this documentation?

Comment (fynmanoj) [reply src/main/java/org/mifos/processor/bulk/api/CallbackController.java:29]: /batchtransactions?command=authorised

Comment (danishjamal104) [reply src/main/java/org/mifos/processor/bulk/api/CallbackController.java:29]: Mentioned in backlog ticket **https://fynarfin.atlassian.net/browse/PHEE-449**

File: src/main/resources/application.yaml (hunk 2 above)
Comment (danishjamal104) [reply src/main/resources/application.yaml]: done

**Review by fynmanoj — APPROVED** [review]

**Pull request merged**